python/cairo-core/src/cairo_core/maths.cairo

```diff
@@ -536,3 +536,69 @@ func felt252_bit_length{range_check_ptr}(value: felt) -> felt {
 
     return bit_length;
 }
+
+// @notice Converts a felt252 to a bit array, little-endian, and outputs to `dst`.
+// @dev Can only convert up to 251 bits included.
+// @returns the length of the bit array.
+func felt252_to_bits_rev{range_check_ptr, bitwise_ptr: BitwiseBuiltin*}(
+    value: felt, len: felt, dst: felt*
+) -> felt {
+    alloc_locals;
+
+    if (len == 0) {
+        return len;
+    }
+    with_attr error_message("felt252_to_bits_rev: len must be < 252") {
+        assert [range_check_ptr] = len;
+        assert [range_check_ptr + 1] = 251 - len;
+        let range_check_ptr = range_check_ptr + 2;
+    }
+
+    let output = &dst[0];
+    // last non-zero bit
+    tempvar last_one: felt;
+    %{ felt252_to_bits_rev %}
+
+    // Start at the first non-zero bit
+    // e.g. if last_one = 3, then initial_acc = 2^3 - 1 = 7,
+    //      if last_one = 0, then initial_acc = 2^0 - 1 = 0
+    tempvar initial_acc = pow2(last_one) - 1;
+    tempvar current_len = last_one;
+    tempvar acc = initial_acc;
+
+    loop:
+    let current_len = [ap - 2];
+    let acc = [ap - 1];
+    let len = [fp - 4];
+
+    // loop stop condition: current_len == len
+    let is_done = is_zero(len - current_len);
+    jmp end if is_done != 0;
+
+    // Check if the bit is valid (0 or 1)
+    let bit = output[current_len];
+    with_attr error_message("felt252_to_bits_rev: bits must be 0 or 1") {
+        assert bit * bit = bit;
+    }
+
+    let pow = pow2(current_len);
+    tempvar shifted_bit = bit * pow;
+    tempvar current_len = current_len + 1;
+    tempvar acc = acc + shifted_bit;
```

Comment:
can avoid access to pow = pow2(len) by multiplying by 2 recursively instead : 

word = b0 + b1 *2 + b2 *2 ^2 + b3 * b^3
word = b0 + 2 * ( b1 + 2 * (b2 + 2 * ( b3)))